src/process.rs

```diff
@@ -14,6 +14,16 @@ pub fn process(
     booking: MutexGuard<ReplyBooking>,
     msg: &Message,
 ) -> Result<Segments> {
+    let delimiter = if let Some(s) = option_env!("HITHIT_BOT_DELIMITER_BUILD") {
+        s.chars().next().unwrap_or('^')
+    } else {
+        std::env::var("HITHIT_BOT_DELIMITER")
+            .unwrap_or("^".to_string())
+            .chars()
+            .next()
+            .unwrap_or('^')
+    };
```

Comment:
This code checks the build time set prefix first, then check runtime env var. IMO the priority should be reversed.

And the prefix is evaluated each time a msg is processed. How about calculate it once and store it in a static variable? In this case， once_cell may help.

And you may simplify this code by `map_or` and `and_then` (or something like that) combinators.

Hint: clippy helps sometimes.